src/lib.rs

```diff
@@ -78,7 +78,10 @@ pub struct RangeInclusive {
 impl RangeInclusive {
     /// Create a new RangeInclusive object.
     pub fn new(start: u64, end: u64) -> Result<Self> {
-        if start >= end {
+        // The length of the interval [0, u64::MAX] is u64::MAX + 1 which does
+        // not fit in a u64::MAX, hence we return `Error::InvalidRange` when
+        // there is an attempt to use that range.
+        if start >= end || (start == 0 && end == u64::MAX) {
```

Comment:
This is not in the scope of the current PR, but was just wondering if we disallow singleton intervals (i.e. `[a,a]` by convention (hence the `start >= end` error convention instead of `start > end`), and if so do we always expect to need intervals with more than one element? Totally fine to address this in the future if there's something to address.